In [2]:
import numpy as np
import joblib
import import_ipynb
from train_model import unpickled_model_gradient

### Sugestão de preço

In [3]:
pathname = './gradient_boosting.pkl'
joblib.dump(unpickled_model_gradient, pathname)

['./gradient_boosting.pkl']

In [4]:
load_model = joblib.load(pathname)

In [5]:
new_value = np.array([[40.75362, -73.98377]])
price = load_model.predict(new_value)
print('Preço: {:.2f}'.format(price[0]))

Preço: 273.49


Para um apartamento com as seguintes características:

- 'id': 2595
- 'nome': 'Skylit Midtown Castle',
- 'host_id': 2845,
- 'host_name': 'Jennifer',
- 'bairro_group': 'Manhattan',
- 'bairro': 'Midtown',
- 'latitude': 40.75362,
- 'longitude': -73.98377,
- 'room_type': 'Entire home/apt',
- 'price': 225,
- 'minimo_noites': 1,
- 'numero_de_reviews': 45,
- 'ultima_review': '2019-05-21',
- 'reviews_por_mes': 0.38,
- 'calculado_host_listings_count': 2,
- 'disponibilidade_365': 355

A sugestão de preço seria de US 273.49.